lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -320,7 +321,31 @@ def _get(
             query = query.filter(Dataset.name.contains(json["name"]))
         if "filter" in json:
             query = self.filter_query(json["filter"], query)
-        query = self._build_sql_query(json.get("owner"), json.get("access"), query)
+
+        # The "mine" filter allows queries for datasets that are (true) or
+        # aren't (false) owned by the authenticated user. In the absense of
+        # the "mine" filter, datasets are selected based on visibility to the
+        # authenticated user and other explicit filters including "owner" and
+        # "access".
+        if "mine" in json:
+            if "owner" in json:
+                raise APIAbort(
+                    HTTPStatus.BAD_REQUEST,
+                    "'owner' and 'mine' filters cannot be used together",
```

Comment:
In theory, yeah, `?mine&owner=<my-username>` wouldn't be a problem, but `?mine=false&owner=<my-username>` is self contradictory, and at best the combination is redundant and doesn't make much sense. I'd rather just refuse it rather than trying to be fancy about it.